TU-Agent/tygron.mod2g

```diff
@@ -5,8 +5,17 @@ use somegoal as goals.
 exit=nogoals.
 
 module helloWorld {
-	% find a function to build a road (always 1 floor)
-	if bel(functions(FS), member([Name,Id,L],FS), member('ROAD',L)) 
-		then building_plan_construction(Id, 1, square(10,10,200,10)).
-
-}
+	% take down the buildings that we have just built
+	if bel(buildings([building(Id, Name, List, Status)])) 
+        then building_plan_demolish(Id) + delete(buildings([building(Id, Name, List, Status)])) + delete(havebuilding) + adopt(havebuilding).
+	
+	% find a function to build a student building
+	 if bel(functions(FS), member([NAME,Id,L],FS), member('EDUCATION',L), land(1)) 
+		then building_plan_construction(Id, 3, square(10,10,200,10)) + insert(built(1)).
```

Comment:
I think it is better to make clear what you have built. So for example built(education_normalroom) instead of built(1).